src/libstd/sys/unix/fs.rs

```diff
@@ -985,7 +994,47 @@ pub fn copy(from: &Path, to: &Path) -> io::Result<u64> {
         }
     }
 
-    let (reader, writer, _, writer_metadata) = open_and_set_permissions(from, to)?;
+    // MacOS prior to 10.12 don't support `fclonefileat`
+    // We store the availability in a global to avoid unnecessary syscalls
+    static HAS_FCLONEFILEAT: AtomicBool = AtomicBool::new(true);
+    syscall! {
+        fn fclonefileat(
+            srcfd: libc::c_int,
+            dst_dirfd: libc::c_int,
+            dst: *const libc::c_char,
+            flags: libc::c_int
+        ) -> libc::c_int
```

Comment:
Missing `;`?

Reply:
I don't think the `syscall!` macro expects a `;`

https://github.com/rust-lang/rust/blob/df738ac1b2c7094d1d2d17a6434d07a63e7d3f2d/src/libstd/sys/unix/weak.rs#L88

Reply:
You're right! It does look odd though.